src/realm/sync/instruction_replication.cpp

```diff
@@ -611,51 +611,62 @@ void SyncReplication::set_clear(const CollectionBase& set)
     }
 }
 
-void SyncReplication::dictionary_update(const CollectionBase& dict, const Mixed& key, const Mixed& value)
+void SyncReplication::dictionary_update(const CollectionBase& dict, const Mixed& key, const Mixed* value)
 {
     // If link is unresolved, it should not be communicated.
-    if (value.is_unresolved_link()) {
+    if (value && value->is_unresolved_link()) {
         return;
     }
 
-    if (select_collection(dict)) {
-        Instruction::Update instr;
-        REALM_ASSERT(key.get_type() == type_String);
-        populate_path_instr(instr, dict);
+    Instruction::Update instr;
+    REALM_ASSERT(key.get_type() == type_String);
+
+    const Table* source_table = dict.get_table().unchecked_ptr();
+    auto col = dict.get_col_key();
+    ObjKey obj_key = dict.get_owner_key();
+    if (source_table->is_additional_props_col(col)) {
+        // Here we have to fake it and pretend we are setting/erasing a property on the object
+        if (!select_table(*source_table)) {
+            return;
+        }
+
+        populate_path_instr(instr, *source_table, obj_key, {key.get_string()});
+    }
+    else {
+        if (!select_collection(dict)) {
+            return;
+        }
+
+        populate_path_instr(instr, *source_table, obj_key, dict.get_short_path());
         StringData key_value = key.get_string();
         instr.path.push_back(m_encoder.intern_string(key_value));
-        instr.value = as_payload(dict, value);
-        instr.is_default = false;
-        emit(instr);
     }
+    if (value) {
+        instr.value = as_payload(*source_table, col, *value);
+    }
+    else {
+        instr.value = Instruction::Payload::Erased{};
```

Comment:
Is this something the server is aware of? (namely that we're sending an Update with Erased payload when an additional property is removed cc @tkaye407)

Reply:
This was what we concluded in one of the check-in meetings.